### ISSUE TITLE
User should be able to define Deployment Strategy: RollingUpdate, Recreate

### ISSUE THREAD

**Issue by ytimocin** [radius-project/radius#6941]
## Overview of feature request
As the title says, Radius user/operator should be able to define the Deployment Strategy of a Deployment: **RollingUpdate**, **Recreate**.

## Acceptance criteria
As a user, I should be able to set the DeploymentStrategy for my Container.

## Additional context

[AB#10734](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10734)

**Comment by rynowak** [comment]
@ytimocin - is this something we need to add to the Radius `Applications.Core/containers` spec? Can you say a little more about how you expect this to be used?

**Comment by ytimocin** [comment]
Yes, we can add this to the **Applications.Core/containers** spec. This could be used to avoid the situation (**if needed**) where two different versions of the application are running simultaneously on the Cluster due to **RollingUpdate** being the default Deployment Strategy.

**Comment by rynowak** [comment]
What would this mean if the user is not deploying to Kubernetes? Can you share more about the scenarios where this is important?

my take: It feels like this is a Kubernetes-specific operational behavior rather than a developer concern. We provide a few ways to 'punch through' the abstraction that you could use to configure this setting without us adding it to `Applications.Core/containers`.

**Comment by ytimocin** [comment]
Some scenarios could include like an internal service that consume message queues that are idempotent and can be upgraded all at once. Another could be where there is a Database migration and RollingUpdate could create a problem.

I agree this is Kubernetes specific and haven't given a lot of thought on a case where we are not deploying to Kubernetes. Happy to brainstorm together if you would like. If we already have some ways to make this happen on a cluster then it should be good enough I guess.
